app/src/core/components/SubmitButton.tsx

```diff
@@ -1,30 +1,31 @@
-import { Button, ButtonTypeMap, CircularProgress, ExtendButtonBase } from '@mui/material';
-import { MouseEvent, PropsWithChildren } from 'react';
+import LoadingButton from '@mui/lab/LoadingButton';
+import { ButtonTypeMap } from '@mui/material';
+import { MouseEvent, PropsWithChildren, useState } from 'react';
 import { animated, useSpring } from 'react-spring';
+import Result, { isErr, isOk } from 'true-myth/result';
 
 const wobbleVolume = 5;
-const AnimatedButton = animated(Button);
+const AnimatedButton = animated(LoadingButton);
 
-export interface SubmitButtonProps extends ExtendButtonBase<ButtonTypeMap<{}, 'button'>> {
+type MuiButtonProps = ButtonTypeMap['props'];
+export interface SubmitButtonProps extends MuiButtonProps {
     disableErrorState?: boolean;
     disableSuccessState?: boolean; // TODO
     iconSize?: number;
-    isLoading?: boolean;
-    onClick?: (event: MouseEvent<HTMLButtonElement, Event>) => Promise<unknown>;
+    onClick?: (event: MouseEvent<HTMLButtonElement, Event>) => Promise<Result<unknown, unknown>>;
+    type: 'button' | 'submit';
 }
 
 export default function SubmitButton({
     children,
     disableErrorState,
     disableSuccessState,
     iconSize,
-    isLoading,
```

Comment:
🤔 I think a `loadingOverride` or something would be fitting here. Specifically thinking about the `authkit` hook or similar apollo functions where - while you get a function that returns a promise - doesn't necessarily reflect the nature of what you're actually waiting for as a user.

Reply:
I suppose that's nit-ish though. That can easily be added when the need arises

Reply:
I do agree having an override to turn the loading state on/off from outside the button would be nice (tho, would just call it `isLoading`). As I can see scenarios like that, or in some niche cases where you may trigger the "submitting" state programmatically via some other event/action thats not necessarily via clicking the button itself.